src/Data/Record/TH/CodeGen.hs

```diff
@@ -293,6 +294,21 @@ genRecordVal opts r@Record{..} mkFn = do
     qNoInlineUnsafeCo :: N.Name 'VarName 'N.Unique -> Q Exp
     qNoInlineUnsafeCo x = [| noInlineUnsafeCo $(N.varE x) |]
 
+genConstructInstance :: Options -> Record () -> Q [Dec]
+genConstructInstance opts r@Record{recordFields} = sequence
+  [ instanceD
+      (cxt [])
+      [t| Construct $recordType |]
+      [ tySynInstD (tySynEqn Nothing [t| FieldsTuple $recordType |] fieldsTupleT)
+      , genRecordVal opts r $ \pats value ->
+          let fieldsTupleP = mkTupleP id (nest DefaultGhcTupleLimit pats)
+           in funD 'construct [clause [tildeP wildP, fieldsTupleP] (normalB value) []]
```

Comment:
We need `tildeP` here because of `Strict` extension on target side

Reply:
Ah, well-spotted sir! I hadn't thought about that.